server/channelserver/handlers_quest.go

```diff
@@ -275,7 +275,38 @@ func makeEventQuest(s *Session, rows *sql.Rows) ([]byte, error) {
 	}
 	bf.WriteUint8(questType)
 	if questType == 9 {
-		bf.WriteBool(false)
+		var stamps int
+		var amount int = 1
+		var deadline time.Time
+		err := s.server.db.QueryRow(`SELECT COUNT(*) FROM campaign_state WHERE campaign_id = (
+			SELECT campaign_id
+			FROM campaign_entries
+			WHERE item_type = 9
+			AND item_no = $1
+		) AND character_id = $2`, questId, s.charID).Scan(&stamps)
+		err2 := s.server.db.QueryRow(`SELECT stamp_amount, (
+			SELECT deadline
+			FROM campaign_entries
+			WHERE item_type = 9
+			AND campaign_id = campaigns.id
+		) AS deadline
+		FROM campaigns
+		WHERE id = (
+			SELECT campaign_id
+			FROM campaign_entries
+			WHERE item_type = 9
+			AND item_no = $1
+		)`, questId).Scan(&amount, &deadline)
+
+		if stamps >= amount && deadline.After(time.Now()) {
+			bf.WriteBool(true)
+
+		} else if err == nil || err2 == nil {
+			bf.WriteBool(false)
+		} else {
+			bf.WriteBool(false)
+
+		}
```

Comment:
The code and formatting here is strange. E.g: Random new line after the first condition and double else condition that seems to return false either way?? That could be simplified.